pkg/node/fsm/ng_state.go

```diff
@@ -205,6 +207,110 @@ func (a *NGState) Block(peer peer.Peer, block *proto.Block) (State, Async, error
 	return newNGState(a.baseInfo), nil, nil
 }
 
+func (a *NGState) BlockEndorsement(blockEndorsement *proto.EndorseBlock) (State, Async, error) {
+	endorsedBlockID := blockEndorsement.EndorsedBlockID
+	endorsedMicroBlock, found := a.baseInfo.MicroBlockCache.GetBlock(endorsedBlockID)
+	if !found || endorsedMicroBlock == nil {
+		return a, nil, a.Errorf(fmt.Errorf(
+			"failed to find the microblock that was endorsed (block ID: %s)",
+			endorsedBlockID.String(),
+		))
+	}
+	top := a.baseInfo.storage.TopBlock()
+	if top.Parent != endorsedMicroBlock.Reference {
+		err := errors.Errorf("endorsed Block ID '%s' must match the parent's block ID '%s'",
+			endorsedMicroBlock.Reference.String(), top.BlockID().String())
+		return a, nil, proto.NewInfoMsg(err)
+	}
+
+	// TODO verify individual endorsements (signatures) using the endorser's public key.
+	activationHeight, actErr := a.baseInfo.storage.ActivationHeight(int16(settings.DeterministicFinality))
+	if actErr != nil {
+		return a, nil,
+			proto.NewInfoMsg(errors.Errorf("failed to get DeterministicFinality activation height, %v", actErr))
+	}
+	height, heightErr := a.baseInfo.storage.Height()
```

Comment:
I think if current height is lower then activation height we should ignore such network messages or even close a connection with the sender.

Reply:
Block endorsement must be ignored (according to the spec) before`DeterministicFinality` feature activation.